cmd/snap/cmd_run.go

```diff
@@ -133,15 +135,32 @@ func createOrUpdateUserDataSymlink(info *snap.Info, usr *user.User) error {
 			// TRANSLATORS: %v the error message
 			return fmt.Errorf(i18n.G("cannot read symlink: %v"), err)
 		}
+
 		if currentSymlinkValue == wantedSymlinkValue {
 			break
 		}
+
 		if !os.IsNotExist(err) {
-			_ = os.Remove(currentActiveSymlink)
+			// We may be racing with other instances of snap-run that try to do the same thing
+			// If the symlink is already removed then we can ignore this error and try again.
+			err = os.Remove(currentActiveSymlink)
+			if os.IsNotExist(err) {
+				time.Sleep(5 * time.Millisecond)
```

Comment:
There's no need to introduce artificial delays here. In the vast majority of cases this will be run by a single process, and otherwise they'll both be racing to do the same action, and succeeding.